providers/keycloak.go

```diff
@@ -63,12 +66,42 @@ func (p *KeycloakProvider) SetGroup(group string) {
 	p.Group = group
 }
 
+func (p *KeycloakProvider) SetRoles(roles []string) {
+	p.Roles = roles
+}
+
+func ExtractRolesFromClaims(claims map[string]interface{}) []string {
```

Comment:
Can you add Unit Tests for this new method? (And add new cases for the additions you made to existing methods)

Reply:
Sure thing, would it make sense to change [auth_test.go](https://github.com/oauth2-proxy/oauth2-proxy/blob/master/providers/auth_test.go#L13) in order to allow passing an access token when calling `CreateAuthorizedSession`? Since the roles are extracted from the access token so it needs to be valid otherwise token parsing will fail.

Reply:
I've added two more tests.

Reply:
I've edited the token in auth_test.go in order to contain a valid token instead of the dummy that was being used. It's currently hardcoded which is not exactly ideal, but if you'd like I can add another method to create the token (that can then be called from the tests in order to create and then set it).

Reply:
Does this function need to be exported?

Reply:
It does not, I unexported it :)